### DISCUSSION TITLE
The public MeshCentral server is shutting down August 12th 2023.

### DISCUSSION THREAD

**Discussion by Ylianst** [Ylianst/MeshCentral#5257] [Announcements]
So, as you guy already know, Intel as ended support for this project last November when the people working on MeshCentral, MeshCommander and other open source tools got laid-off. MeshCentral is mostly on GitHub and so there is no cost to keeping the project online except for the public MeshCentral.com server which currently runs on Microsoft Azure and costs around 280$/month. The cost of the instance is pretty cheap, around 50$/mo or so, but it's the traffic cost that is really high as there are a lot of people running remote desktop sessions, traffic routing, etc.

I don't really want to deal with that monthly cost and also the time and risk involved in running such a server publicly. There are a lot of ways such a public server can be abused and so, shutting it down seems like a sensible thing to do. I do want to note that if your running your own MeshCentral server, this does not impact you at all and MeshCentral will continue to be on GitHub and I will keep the MeshCentral.com main page running. I may move the MeshCentral.com page to a GitHub Sites at a later point which would really make the hosting costs go to zero. This will also free me up to be more of a developer and less of a IT admin, which is something I look forward to.

Hope this makes sense. I don't expect this will impact anyone that was participating on GItHub much as I imagine people around here are all running their own instances which is the right thing to do from a security perspective.

**Comment by silversword411** [comment]
@Ylianst Would you mind sharing the config customizations you did for the public server if it's not much work?

**Reply by Ylianst** [reply]
@silversword411 Absolutely! Working on getting that now. I will XXXX out the secrets of course. I should have shared that before, no one asked.

**Comment by Ylianst** [comment]
Here it is, pretty standard mostly except for "FireBase", the "info" site.

```
{
  "settings": {
    "Cert": "meshcentral.com",
    "MongoDb": "mongodb://127.0.0.1:27017/meshcentral",
    "MongodbBulkOperations": true,
    "WANonly": true,
    "IgnoreAgentHashCheck": true,
    "SessionKey": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "WebRTC": false,
    "ClickOnce": true,
    "RelayDns": [ "relay1.meshcentral.com", "relay2.meshcentral.com", "relay3.meshcentral.com" ],
    "AgentBlockIP": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "AgentCoreDump": true,
    "AgentCoreDumpUsers": [ "user//admin" ],
    "_WebRtcConfig": { "iceServers": [ { "urls": "stun:stun.services.mozilla.com" }, { "urls": "stun:stun.l.google.com:19302" } ] },
    "NpmPath": "/home/ec2-user/.nvm/versions/node/v15.11.0/bin/node /home/ec2-user/.nvm/versions/node/v15.11.0/bin/npm",
    "AllowHighQualityDesktop": false,
    "UserBlockedIp": [ "XXXXXXXXXXXXXXXXXXXXXXXXXXX" ]
  },
  "domains": {
    "": {
      "siteStyle": 2,
      "NewAccounts": 1,
      "rootRedirect": "https://meshcentral.com/info/",
      "UserConsentFlags": 71,
      "httpHeaders": {
        "Strict-Transport-Security": "max-age=31536000",
        "x-frame-options": "SAMEORIGIN"
      },
      "welcomeText": "<b>This server will be shutting down on Auguest 12th 2023.</b>",
      "Footer": "<b>This server will be shutting down on Auguest 12th 2023.</b>",
      "PasswordRequirements": { "min": 8, "max": 128, "upper": 1, "lower": 1, "numeric": 1, "nonalpha": 1 },
      "yubikey": { "id": "XXXXXXXXXXX", "secret": "XXXXXXXXXXXXXXXXXXXXXXXXXX/4=" },
      "Limits": {
        "MaxSingleUserSessions": 10
      },
      "Redirects": {
        "sdfklsdgsdg9e4tdrsjktgversion": "~showversion"
      },
      "userSessionIdleTimeout": 5
    },
    "info": {
      "share": "/home/azureuser/meshcentral-infosite/",
      "httpheaders": {
          "Referrer-Policy": "no-referrer",
          "X-XSS-Protection": "1; mode=block",
          "X-Content-Type-Options": "nosniff",
          "Permissions-Policy": "interest-cohort=()",
          "Content-Security-Policy": "default-src 'none' www.youtube.com; font-src 'self'; script-src 'self' 'unsafe-inline'; img-src 'self'; media-src 'self'; style-src 'self' 'unsafe-inline'"
      }
    }
  },
  "letsencrypt": {
    "email": "XXXXXXXXXXXXXXXXXXXX@hotmail.com",
    "names": "meshcentral.com,www.meshcentral.com,relay1.meshcentral.com,relay2.meshcentral.com,relay3.meshcentral.com",
    "production": true
  },
  "smtp": {
    "from": "ylianst@gmail.com",
    "auth": {
      "clientid": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX.apps.googleusercontent.com",
      "clientsecret": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
      "refreshtoken": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    }
  },
  "firebase": {
    "senderId": "XXXXXXXXXXXXXXXXXXXXXX",
    "serverKey": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "pushRelayServer": true,
    "relayServer": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
  }
}
```

**Comment by lixxdee** [comment]
@Ylianst Why don't you want to open donations on github to support your project, including public service MeshCentral?

**Reply by si458** [reply]
this already exists - https://github.com/sponsors/Ylianst
i have my own sponsors page too to try and help community repos - https://github.com/sponsors/si458

**Reply by Ylianst** [reply]
In my case I don't need the money. It's really about my time. @si458 has been doing wonderful work and I would certainly support this work. Running the public server is a bunch of work and it's also risk for the people that connect their computers to that server. Really, running your own server is a lot more secure. I am also worried that people have been using the public MeshCentral server to do bad things, I keep getting requests from people asking to remove the privacy bar when doing remote desktop of the public server and I wonder. In any case, it's the right thing to do. It does not stop anyone from launching their own MeshCentral public server, but I would be very careful who you trust your computers to.

**Comment by bluestreak-vl** [comment]
First, thank you for all your work.

I have been using the public server with all Windows clients so far. Is there way to easily update the configuration on them to transition them to a new server? Maybe a registry edit?

Thanks

**Reply by jsastriawan** [reply]
Here is what I came into my mind based old trick I did when I migrated old meshcentral1 to meshcentral 2 (current meshcentral)

installedpath\meshagent.exe -fulluninstall && newinstaller\meshagent.exe -fullinstall

Please test it first if it is working before running it on production.

**Reply by si458** [reply]
I think in this case because its the public server, the only thing u can do is create ur own server,
Then customise the agent and install the new agent on the computers then uninstall the old public agent

**Reply by bluestreak-vl** [reply]
> Here is what I came into my mind based old trick I did when I migrated old meshcentral1 to meshcentral 2 (current meshcentral)
> 
> installedpath\meshagent.exe -fulluninstall && newinstaller\meshagent.exe -fullinstall
> 
> Please test it first if it is working before running it on production.

I found that the two uninstall/install commands in one line didn't work but "newinstaller\meshagent.exe -fullinstall" worked by itself. During the install process, it would uninstall the old version too.

Works from the Actions -> Run Command window and even the interactive Terminal window.

**Reply by bluestreak-vl** [reply]
Another way is to update the meshagent.msh file with contents of the same file from the new new server and then restarting the meshagent service using "meshagent.exe restart". 
Of course this means downloading a new mesh agent executable from the new server to get the contents of the .msh file.

**Comment by marclaporte** [comment]
Relevant: https://meshcentral2.blogspot.com/2023/10/meshcentral-windows-arm64-nodejs-v11.html

I urge everyone to join me in sponsoring Simon Smith (si458)!
https://github.com/sponsors/si458

We need your contributions: time, code or money!
https://github.com/Ylianst/MeshCentral/discussions/5540
